tests/test.VmAgentRealVmadm.js

```diff
@@ -841,6 +842,136 @@ test('Real vmadm, fake VMAPI: validate DNI', function _test(t) {
+        function _createDoNotInventoryVm1(arg, cb) {
+            // creating a VM with DNI should not result in it being in the
+            // initial update.
+            vmadm.create(payload1, function _vmadmCreateCb(err, info) {
+                t.ifError(err, 'create DNI VM1');
+                if (!err && info) {
+                    t.ok(info.uuid, 'new DNI VM1 has uuid: ' + info.uuid);
+                }
+                cb(err);
+            });
+        }, function _startVmAgent(arg, cb) {
+            t.ok(config.server_uuid, 'new CN ' + config.server_uuid);
+            vmAgent = new VmAgent(config);
+            vmAgent.start();
+            cb();
+        }, function _waitInitialUpdateVms(arg, cb) {
+            // Wait for VmAgent init and it'll send the initial PUT /vms, these
+            // are real VMs on the node because we're not faking vmadm.
+            coordinator.once('vmapi.updateServerVms',
+                function _onUpdateVms(vmobjs /* , server_uuid */) {
+                    var keys = Object.keys(vmobjs);
+
+                    t.ok(true, 'saw PUT /vms: (' + keys.length + ')');
+                    keys.forEach(function _putVmToVmapi(vm) {
+                        // ignore updates from VMs that existed when we started
+                        mocks.Vmapi.putVm(vmobjs[vm]);
+                    });
+                    cb();
+                }
+            );
+        }, function _createDoNotInventoryVm2(arg, cb) {
+            // creating a VM with DNI should not result in an update.
+            vmadm.create(payload2, function _vmadmCreateCb(err, info) {
+                t.ifError(err, 'create DNI VM2');
+                if (!err && info) {
+                    t.ok(info.uuid, 'new DNI VM2 has uuid: ' + info.uuid);
+                }
+                cb(err);
+            });
+        }, function _destroyVm1(arg, cb) {
+            //  With all modifications complete, we now delete the VM
+            vmadm.delete({
+                include_dni: true,
+                log: config.log,
+                uuid: payload1.uuid
+            }, function _vmadmDeleteCb(err) {
+                t.ifError(err, 'delete VM: '
+                    + (err ? err.message : 'success'));
+                cb(err);
+            });
+        }, function _destroyVm2(arg, cb) {
+            //  With all modifications complete, we now delete the VM
+            vmadm.delete({
+                include_dni: true,
+                log: config.log,
+                uuid: payload2.uuid
+            }, function _vmadmDeleteCb(err) {
+                t.ifError(err, 'delete VM: '
+                    + (err ? err.message : 'success'));
+                cb(err);
+            });
+        }, function _waitDestroyVm(arg, cb) {
+            setTimeout(function _waitedDestroyVm() {
+                t.ok(true, 'waited ' + WAIT_DNI_DELETE + ' ms after delete');
+                cb();
+            }, WAIT_DNI_DELETE);
```

Comment:
We're waiting in case for example the periodic watcher (which does a vmadm lookup every second for this test, but where vmadm lookup can take a couple seconds occasionally) sees a delete. We want to make sure that none of the updates are for this VM. I'm not sure how we'd make a sentinel for that really.